lmfit/parameter.py

```diff
@@ -188,6 +188,20 @@ def __setstate__(self, state):
         # then add all the parameters
         self.add_many(*state['params'])
 
+    def eval(self, expr):
+        """Evaluate a statement using the asteval Interpreter.
+        Parameters
+        ----------
+        expr : string
+            An expression containing parameter names and other symbols
+            recognizable by the asteval Interpreter.
+
+        Returns
+        -------
+           The result of the expression.
```

Comment:
the return type is ```float```, so please add that before the description:
```
float
    The result of the expression.
```

Reply:
@reneeotten, this might be a bit nitpicky, but the return type of eval can be pretty much anything. A few dummy examples: params.eval("bool(1.0)") or params.eval("1 + 2j").

Should I go ahead with using 'float' since it's the most likely? Or, is there a better way to address the return type?

Reply:
ah, good point... I guess I only looked at the example you added (where the return values was ```float```). I am not sure what to make of the return type then, so by lack of a better idea we can leave it as is. Should this new method be added to the documentation (@newville)?

Reply:
Ah, yes, please add method to doc.  

I guess that the value of a Parameter does not *need* to be `float` unless it is really a variable parameter used in the actual fit.  That is, I suppose it would be OK for derived parameters to be complex, bools, or ints.

Reply:
@zobristnicholas can you please add ```.. automethod:: eval``` to ```doc/parameters.rst``` near line 76 after ```.. automethod:: dump```? I just tried that, and it doesn't parse the docstring correctly because it needs to be another empty line as in:
```
         Returns
         -------
            The result of the expression.

         """
```

Please squash commits with changes to the docstring together with the first one ("added eval method to Parameters class"); you can add a new commit for adding it to the documentation.

Reply:
@reneeotten, how do those changes look?